### PR TITLE
Elasticsearch client.deleteByQuery params update for 5.x

### PR DESCRIPTION
**Pull request by mlamp**
Please fill in this template.

- [x] Make your PR against the `master` branch.
- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped#common-mistakes).
- [x] Run `tsc` without errors.
- [x] Run `npm run lint package-name` if a `tslint.json` is present.

Select one of these and delete the others:
If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://www.elastic.co/guide/en/elasticsearch/client/javascript-api/current/api-reference.html#api-deletebyquery
- [x] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "../tslint.json" }`.
- [x] Increase the version number in the header if appropriate.

### CODE REVIEW

File: elasticsearch/index.d.ts [hunk 1]
```diff
@@ -234,16 +235,37 @@ declare module Elasticsearch {
...
+        lenient?: boolean;
+        lowercaseExpandedTerms?: boolean;
+        preference?: string;
+        q?: string;
+        routing?: string | string[] | boolean;
+        scroll?: string;
+        searchType?: "query_then_fetch" | "dfs_query_then_fetch";
+        searchTimeout?: TimeSpan;
+        size?: number;
+        sort?: NameList;
```
Comment (pushplay) [inline elasticsearch/index.d.ts]: Why not include `_source`, `_sourceExclude` and `_sourceInclude`?

Comment (mlamp) [inline elasticsearch/index.d.ts]: Done.

Comment (pushplay) [inline elasticsearch/index.d.ts]: Thanks!

👍

**Comment by pushplay** [comment]
👍

File: elasticsearch/index.d.ts (hunk 1 above)
Comment (mlamp) [inline elasticsearch/index.d.ts]: Cheers, np! :)

**Comment by mlamp** [comment]
Updated my PR, df and from are optional parameters.

**Pull request merged**